src/System.Text.Json/src/System/Text/Json/Serialization/WriteStackFrame.cs

```diff
@@ -139,20 +140,28 @@ public void EndArray()
             PopStackOnEndCollection = false;
         }
 
+        // AggressiveInlining used although a large method it is only called from one location and is on a hot path.
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
         public void NextProperty()
         {
             EndProperty();
 
             if (PropertyEnumeratorActive)
             {
-                if (PropertyEnumerator.MoveNext())
+                int len = JsonClassInfo.PropertyCacheArray.Length;
+                if (PropertyEnumeratorIndex < len)
                 {
+                    if ((PropertyEnumeratorIndex == len - 1) && JsonClassInfo.DataExtensionProperty != null)
```

Comment:
This is hit 100% of the time when we can an extension property.

Reply:
To clarify, I was talking about test coverage. It looks like this change introduced test gaps that we should fix. Can you please add tests so that all new branches/lines of code being introduced in this PR are covered?

For example, `NextProperty` is no longer 100% covered:
![image](https://user-images.githubusercontent.com/6527137/65569028-d7881000-df10-11e9-9046-c6b432ccf8fd.png)

Reply:
Thanks. I will send a new PR to clean this up.